consensus/consensus-types/src/block_retrieval.rs

```diff
@@ -9,20 +9,37 @@ use serde::{Deserialize, Serialize};
 use short_hex_str::AsShortHexStr;
 use std::fmt;
 
-pub const MAX_BLOCKS_PER_REQUEST: u64 = 10;
+// this number is recommended to be greater than backpressure limit
+// (so block retrievals can be done in a single request)
+// but it should not be too large as the response size is bounded.
+pub const MAX_BLOCKS_PER_REQUEST: u64 = 40;
```

Comment:
did some calculation I don't think we can increase the limit here, we should follow up with test as well.
a single txn is ~400 bytes, a block could have 1k txns, so 40 blocks can take ~40 * 1000 * 400 / 1024 / 1024 b = 15mb >> 8mb frame limit

Reply:
I see. I'll add a test later.